test/decl/protocol/special/comparable/comparable_unsupported_type.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %target-typecheck-verify-swift -verify-ignore-unknown
+
+// Automatic synthesis is only supported for enums for now.
+
+struct NotComparableStruct: Comparable {
+  // expected-error@-1 {{type 'NotComparableStruct' does not conform to protocol 'Comparable'}}
+  // expected-note@-2 {{automatic synthesis of 'Comparable' is not supported for structs}}
+  var value = 0
+}
+
+class NotComparableClass: Comparable {
+  // expected-error@-1 {{type 'NotComparableClass' does not conform to protocol 'Comparable'}}
+  // expected-note@-2 {{automatic synthesis of 'Comparable' is not supported for classes}}
+  // expected-error@-3 {{type 'NotComparableClass' does not conform to protocol 'Equatable'}}
+  // expected-note@-4 {{automatic synthesis of 'Equatable' is not supported for classes}}
+  var value = 1
+}
+
+// Make sure we don't emit the note for enums.
+
+enum NotComparableEnum: Int, Comparable {
+  // expected-error@-1 {{type 'NotComparableEnum' does not conform to protocol 'Comparable'}}
+  case value = 2
+}
```

Comment:
Not part of this PR but I find it puzzling (and I didn't realize this before) that such enums do not support automatic synthesis of `Comparable`. 🤔